python-package/lightgbm/basic.py

```diff
@@ -821,6 +833,9 @@ def predict(
         validate_features : bool, optional (default=False)
             If True, ensure that the features used to predict match the ones used to train.
             Used only if data is pandas DataFrame.
+        use_int64 : bool, optional (default=False)
+            If True, support dataset rows more then max(int32_t).
+            Need to build with cmake option USE_DATASET_INT64 for lightGBM.
```

Comment:
I think instead of introducing a new param in a lot of places we can define a new C API function that will return whether LightGBM was compiled with int64 support or not.

Reply:
For example, in XGBoost:
https://github.com/dmlc/xgboost/blob/97c3a80a346828165c57225929657461bcaeeda2/include/xgboost/c_api.h#L65-L72
https://github.com/dmlc/xgboost/blob/8d4038da57cd06070677460d138a648c132665b1/src/c_api/c_api.cc#L59-L112